packages/features/booking-audit/lib/service/booking-audit.integration-test.ts

```diff
@@ -0,0 +1,284 @@
+        email: testAttendeeEmail,
+        name: "Test Attendee",
+        timeZone: "UTC",
+        bookingId: (await prisma.booking.findUnique({ where: { uid: testBookingUid }, select: { id: true } }))!.id,
+      },
+    });
+  });
+
+  afterEach(async () => {
+    // Clean up test data in correct order to respect foreign key constraints
+
+    // Delete audit logs first (references actor)
+    await prisma.bookingAudit.deleteMany({
+      where: { bookingUid: testBookingUid },
+    });
+
+    // Delete actors
+    await prisma.auditActor.deleteMany({
+      where: {
+        OR: [
+          { userUuid: testUserUuid },
+          { email: testAttendeeEmail },
+        ],
+      },
+    });
+
+    // Delete attendees
+    await prisma.attendee.deleteMany({
+      where: { email: testAttendeeEmail },
+    });
+
+    // Delete booking
+    await prisma.booking.deleteMany({
+      where: { uid: testBookingUid },
+    });
+
+    // Delete event type
+    await prisma.eventType.deleteMany({
+      where: { id: testEventTypeId },
+    });
+
+    // Delete users
+    await prisma.user.deleteMany({
+      where: {
+        id: {
+          in: [testUserId, testAttendeeUserId],
+        },
+      },
+    });
+  });
+
+  describe("CREATED action end-to-end flow", () => {
+    it("should create audit record and retrieve it with correct formatting", async () => {
+      // Arrange: Get booking details
+      const booking = await prisma.booking.findUnique({
+        where: { uid: testBookingUid },
+        select: {
+          startTime: true,
+          endTime: true,
+          status: true,
+        },
+      });
+
+      expect(booking).toBeDefined();
+
+      // Create actor using helper
+      const actor = makeUserActor(testUserUuid);
+
+      // Act: Create audit record using TaskConsumer
+      await bookingAuditTaskConsumer.onBookingAction({
+        bookingUid: testBookingUid,
+        actor,
+        action: "CREATED",
+        data: {
+          startTime: booking!.startTime.getTime(),
+          endTime: booking!.endTime.getTime(),
+          status: booking!.status,
+        },
+        timestamp: Date.now(),
+      });
+
+      // Retrieve audit logs
+      const result = await bookingAuditViewerService.getAuditLogsForBooking(
+        testBookingUid,
+        testUserId,
+        testUserEmail
+      );
+
+      // Assert: Verify audit log exists and has correct data
+      expect(result.bookingUid).toBe(testBookingUid);
+      expect(result.auditLogs).toHaveLength(1);
+
+      const auditLog = result.auditLogs[0];
+      expect(auditLog.bookingUid).toBe(testBookingUid);
+      expect(auditLog.action).toBe("CREATED");
+      expect(auditLog.type).toBe("RECORD_CREATED");
+
+      // Verify audit data matches (getDisplayJson returns fields only, not versioned wrapper)
+      const displayData = auditLog.data as { startTime: number; endTime: number; status: string };
+      expect(displayData.startTime).toBe(booking!.startTime.getTime());
```

Comment:
Integration tests are failing

Reply:
Fixed it. last moment change broke it. Thanks !!